File: src/prod/src/ServiceModel/naming/PropertyBatchInfo.cpp
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
// ------------------------------------------------------------

#include "stdafx.h"

namespace Naming
{
    using namespace std;
    using namespace Common;

    PropertyBatchInfo::PropertyBatchInfo()
        : kind_(PropertyBatchInfoType::Invalid)
        , errorMessage_()
        , operationIndex_(FABRIC_INVALID_OPERATION_INDEX)
        , properties_()
    {
    }

    PropertyBatchInfo::PropertyBatchInfo(NamePropertyOperationBatchResult && batchResult)
        : kind_(PropertyBatchInfoType::Invalid)
        , errorMessage_()
        , operationIndex_(FABRIC_INVALID_OPERATION_INDEX)
        , properties_()
    {
        if (batchResult.Error.IsSuccess())
        {
            kind_ = PropertyBatchInfoType::Successful;

            std::vector<NamePropertyResult> properties = batchResult.TakeProperties();
            for (auto itr = properties.begin(); itr != properties.end(); ++itr)
            {
                std::wstring index(StringUtility::ToWString(itr->Metadata.OperationIndex));
                PropertyInfo propInfo = PropertyInfo::Create(std::move(*itr), true);
                properties_.emplace(std::move(index), std::move(propInfo));
            }

            std::vector<NamePropertyMetadataResult> metadata = batchResult.TakeMetadata();
            for (auto itr = metadata.begin(); itr != metadata.end(); ++itr)
            {
                std::wstring index(StringUtility::ToWString(itr->OperationIndex));
                PropertyInfo propInfo = PropertyInfo(std::move(*itr));
                properties_.emplace(std::move(index), std::move(propInfo));
            }
        }
        else
        {
            kind_ = PropertyBatchInfoType::Failed;
            errorMessage_ = batchResult.Error.ErrorCodeValueToString();
            operationIndex_ = batchResult.FailedOperationIndex;
        }
    }

    PropertyBatchInfo::~PropertyBatchInfo()
    {
    }
}

File: src/prod/test/httpgatewaytest/PropertyBatchInfo.cs
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
// ------------------------------------------------------------

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Runtime.Serialization;
using System.Fabric;

namespace System.Fabric.Test.HttpGatewayTest
{
    [Serializable]
    [DataContract]
    class PropertyBatchInfo
    {

        public PropertyBatchInfo()
        {
            Kind = "";
            Properties = new Dictionary<string, PropertyInfo>();
            ErrorMessage = null;
            OperationIndex = -1;
        }

        [DataMember]
        public string Kind
        {
            get;
            set;
        }

        [DataMember]
        public IDictionary<string, PropertyInfo> Properties
        {
            get;
            set;
        }

        [DataMember]
        public string ErrorMessage
        {
            get;
            set;
        }

        [DataMember]
        public long OperationIndex
        {
            get;
            set;
        }
    }
}